crates/aptos-drop-helper/src/lib.rs

```diff
@@ -0,0 +1,108 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+mod metrics;
+
+use crate::metrics::TIMER;
+use aptos_experimental_runtimes::thread_manager::THREAD_MANAGER;
+use aptos_infallible::Mutex;
+use aptos_metrics_core::TimerHelper;
+use once_cell::sync::Lazy;
+use std::sync::mpsc::{channel, Receiver, Sender};
+
+pub static DEFAULT_DROP_HELPER: Lazy<DropHelper> = Lazy::new(|| DropHelper::new("default", 128));
+
+/// A helper to send things to a thread pool for asynchronous dropping.
+///
+/// Be aware that there is a bounded number of concurrent drops, as a result:
+///   1. when it's "out of capacity", `schedule_drop` will block until a slot to be available.
+///   2. if the `Drop` implementation tries to lock things, there can be a potential dead lock due
+///      to another thing being waiting for a slot to be available.
+pub struct DropHelper {
+    name: &'static str,
+    token_tx: Sender<()>,
+    token_rx: Mutex<Receiver<()>>,
+}
+
+impl DropHelper {
+    pub fn new(name: &'static str, max_concurrent_drops: usize) -> Self {
+        let (token_tx, token_rx) = channel();
+        for _ in 0..max_concurrent_drops {
+            token_tx.send(()).expect("Failed to buffer initial tokens.");
```

Comment:
nit: "failed to buffer initial tokens for concurrent dropping "

Reply:
okay... 
I put in an `expect` instead of `unwrap` just to avoid being "nit: expect()"ed, and now I still got nitted and have to make this bullshit 3 lines 😢 :

Reply:
done